core/variable_map.js

```diff
@@ -62,15 +62,15 @@ Blockly.VariableMap.prototype.clear = function() {
  * Rename the given variable by updating its name in the variable map.
  * @param {Blockly.VariableModel} variable Variable to rename.
  * @param {string} newName New variable name.
+ * @param {string=} opt_type The type of the variable to create if variable was
```

Comment:
I don't know.  Possibly as part of undo/redo cycles?

It's explicitly tested by `test_undoRedoRenameVariable_NeitherVariableExists`.

@marisaleung do you remember why?

Reply:
I chatted with Marisa offline.  She thinks it's related to dragging blocks with variables out of the flyout.  It may become a non-problem once we have a VariableModel for every variable reference.